subprojects/core/src/integTest/groovy/org/gradle/api/ExternalScriptExecutionIntegrationSpec.groovy

```diff
@@ -73,4 +77,156 @@ task check {
         expect:
         succeeds 'check'
     }
+
+    @Unroll
+    def "will used cached #source resource when run with --offline"() {
+        given:
+        String scriptName = "script-${System.currentTimeMillis()}.gradle"
+        executer.withDefaultCharacterEncoding("ISO-8859-15")
+
+        and:
+        def scriptFile = file("script.gradle")
+        scriptFile.setText("""println 'loaded external script'""", "UTF-8")
+        server.expectGet('/' + scriptName, scriptFile)
+
+        and:
+        file("init.gradle").createFile()
+        buildFile << """
+            task check {}
+        """
+
+        when:
+        def scriptUri = "http://localhost:${server.port}/${scriptName}"
+        file(sourceFile) << """
+            apply from: '${scriptUri}'
+        """
+
+        then:
+        args('-I', 'init.gradle')
+        succeeds 'check'
+        outputContains 'loaded external script'
+
+        when:
+        server.stop()
+
+        then:
+        args('-I', 'init.gradle')
+        fails 'check'
+        errorOutput.contains("Could not get resource '${scriptUri}'")
+
+        when:
+        scriptFile.setText("""throw new RuntimeException('NOT CACHED')""", "UTF-8")
+        args('--offline', '-I', 'init.gradle')
```

Comment:
This test is designed to ensure that a user can use `--offline` to recover from failure after discovering that the server is offline. I guess we could add support for the case of using `--offline` to eliminate requests altogether.